tests/ui/match_same_arms.rs

```diff
@@ -108,6 +108,36 @@ fn match_same_arms() {
         (None, Some(a)) => bar(a), // bindings have different types
         _ => (),
     }
+
+    let _ = match 42 {
+        42 => 1,
+        51 => 1, //~ ERROR match arms have same body
+        41 => 2,
+        52 => 2, //~ ERROR match arms have same body
+        _ => 0,
+    };
+
+    let _ = match 42 {
+        1 => 2,
+        2 => 2, //~ ERROR 2rd matched arms have same body
+        3 => 2, //~ ERROR 3rd matched arms have same body
+        4 => 3,
+        _ => 0,
+    };
+
+    let try_it = if true {
```

Comment:
This test seems good now. Maybe:
```suggestion
    let _ = if true {
```

Can you move this test to the correct file?

Also you could change one `true` to `1 == 2` or something, to avoid the `clippy::ifs_same_cond` error.

Reply:
"1 == 2" condition throws me the same error for the clippy::ifs_same_cond error.

Reply:
Oh then the lint is too smart :D In that case, could you just allow that lint in the test file?